src/test/java/com/endava/service/MoneyTransferServiceImplIT.java

```diff
@@ -0,0 +1,251 @@
+
+        moneyTransfer.setUser(user);
+
+        MoneyTransfer persistedMoneyTransfer = moneyTransferService.addMoneyTransfer(moneyTransfer);
+        assertNotNull(persistedMoneyTransfer);
+
+        moneyTransferService.deleteMoneyTransferById(persistedMoneyTransfer.getId());
+        assertFalse(moneyTransferDao.exists(persistedMoneyTransfer));
+    }
+
+    @Test
+    public void testUpdateMoneyTransfer() {
+        userDao.persist(user);
+        wherefromDao.persist(wherefrom);
+
+        moneyTransfer.setUser(user);
+
+        MoneyTransfer persistedMoneyTransfer = moneyTransferService.addMoneyTransfer(moneyTransfer);
+        assertNotNull(persistedMoneyTransfer);
+
+        persistedMoneyTransfer.setAmount(10000000000.0);
+
+        MoneyTransfer updatedMoneyTransfer = moneyTransferService.updateMoneyTransfer(persistedMoneyTransfer);
+        assertNotNull(updatedMoneyTransfer);
+
+        assertEquals(persistedMoneyTransfer, updatedMoneyTransfer);
+    }
+
+    @Test
+    public void testGetMoneyTransferById() {
+        userDao.persist(user);
+        wherefromDao.persist(wherefrom);
+
+        moneyTransfer.setUser(user);
+
+        MoneyTransfer persistedMoneyTransfer = moneyTransferService.addMoneyTransfer(moneyTransfer);
+        assertNotNull(persistedMoneyTransfer);
+
+        MoneyTransfer pulledMoneyTransfer = moneyTransferService.getMoneyTransferById(persistedMoneyTransfer.getId());
+        assertNotNull(pulledMoneyTransfer);
+
+        assertEquals(persistedMoneyTransfer, pulledMoneyTransfer);
+    }
+
+    @Test
+    public void testGetMoneyTransfers() {
+        userDao.persist(user);
+        wherefromDao.persist(wherefrom);
+
+        moneyTransfer.setUser(user);
+
+        MoneyTransfer persistedMoneyTransfer = moneyTransferService.addMoneyTransfer(moneyTransfer);
+        assertNotNull(persistedMoneyTransfer);
+
+        List<MoneyTransfer> persistedMoneyTransfers = Arrays.asList(persistedMoneyTransfer);
+
+        List<MoneyTransfer> pulledMoneyTransfers = moneyTransferService.getMoneyTransfers();
+        assertNotNull(pulledMoneyTransfers);
+
+        assertEquals(persistedMoneyTransfers, pulledMoneyTransfers);
+    }
+
+    @Test
+    public void testGetMoneyTransferOn() {
+        Date date = new Date(DATE_ONE);
+
+        userDao.persist(user);
+        wherefromDao.persist(wherefrom);
+
+        moneyTransfer.setUser(user);
+
+        MoneyTransfer persistedMoneyTransfer = moneyTransferService.addMoneyTransfer(moneyTransfer);
+        assertNotNull(persistedMoneyTransfer);
+
+        List<MoneyTransfer> requiredMoneyTransfer = moneyTransferService.getMoneyTransferOn(date);
+        assertNotNull(requiredMoneyTransfer);
+
+        assertEquals(Arrays.asList(persistedMoneyTransfer), requiredMoneyTransfer);
+    }
+
+    @Test
+    public void testGetMoneyTransferLastWeek() {
+        userDao.persist(user);
+        wherefromDao.persist(wherefrom);
+
+        MoneyTransfer newMT = MoneyTransfer.builder()
```

Comment:
externalize this builder.
add a private method:

private MoneyTransfer createNewMoneyTransfer(User user, Wherefrom wherefrom) {
return MoneyTransfer.builder().....
}